src/engine/controls/cuecontrol.cpp

```diff
@@ -1979,9 +2453,27 @@ void HotcueControl::resetCue() {
     // in the code
     m_pCue.reset();
     setPosition(Cue::kNoPosition);
+    setEndPosition(Cue::kNoPosition);
+    setType(mixxx::CueType::Invalid);
+    setStatus(Status::Empty);
 }
 
 void HotcueControl::setPosition(double position) {
     m_hotcuePosition->set(position);
-    m_hotcueEnabled->forceSet(position == Cue::kNoPosition ? 0.0 : 1.0);
+}
+
+void HotcueControl::setEndPosition(double endPosition) {
+    m_hotcueEndPosition->set(endPosition);
+}
+
+void HotcueControl::setType(mixxx::CueType type) {
+    m_hotcueType->forceSet(static_cast<double>(type));
+}
+
+void HotcueControl::setStatus(HotcueControl::Status status) {
+    m_pHotcueStatus->forceSet(static_cast<double>(status));
+}
+
+HotcueControl::Status HotcueControl::getStatus() const {
+    return static_cast<Status>(m_pHotcueStatus->get());
```

Comment:
This fails on MSVC 
..\src\engine\controls\cuecontrol.cpp(2478): error C2440: 'static_cast': cannot convert from 'double' to 'HotcueControl::Status'
..\src\engine\controls\cuecontrol.cpp(2478): note: Conversions between enumeration and floating point values are no longer allowed

Reply:
I pushed a fix that casts to int first. If that works, should I try to fixup the commit that originally introduced that and force-push? I'm worried about losing the review comments.

Reply:
I think there is no need for this. If you not hesitate the work, go ahead. There is IMHO only one issue left before merge.

Reply:
OK. I was just asking because of our policy to have every commit compile (to make bisecting easier). if you don't think is is necessary I won't fixup the commit.